### DISCUSSION TITLE
Where to get older versions of dosemu ppa?

### DISCUSSION THREAD

**Discussion by nelgin** [dosemu2/dosemu2#2226] [General]
I just noticed on my BBS that dosemu2 games. is no longer working. Games just exit right back to the menu. They seem to work in local mode. I've regressed my BBS software back to March and still see the problem so I'm guessing the a regression with dosemu2 somewhere.

I'd like to install the previous version, dosemu2 2.0~pre9-8552-fe5977761+202403162131~ubuntu22.04.1, but I don't see the binary .deb files anywhere. Google throws up nothing.

Does anyone know a place to get this and any related dependencies? I may need to go back a few months. Maybe I am going to have to build the older version myself, which isn't ideal because it wouldn't re-create, exactly, the ppa version.

Thanks,

**Comment by stsp** [comment]
Its nearly impossible to fully build
older dosemu2. And getting it from
PPA won't work anyway, as it would
try the recent deps (fdpp, comcom64),
but they won't work with older dosemu2.
But you can try to use freedos or ms-dos
under the latest one, to exclude the
regression in fdpp or command.com.
After that, you can go back by producing
the reduced builds (`--disable-fdpp` and
disabling dj64 plugin). These builds would
still be able to boot freedos or ms-dos.
This is the only way to roll back the
history, unfortunately.

**Reply by nelgin** [reply]
Right, I would also want to get the dependencies and just remove the latest and install the previous version. I guess there's not an archive of older versions. Maybe someone has copies for some odd reason? Of course, fixing the issue would be better than trying to find which version it was broken in, but unfortunately there's no errors or anything, even with -D+a from what I can tell. 

That said...in an older log I have:

Warning: SERIAL $Id$
registering /dev/pts/1 for serial
DPMI: mem init, mpool is 140468224 bytes at 0x7f7423e00000
Warning: Using DPMI with CPU emulator

Then later
Warning: CONF: dosemu not running on console
CONF: time mode = 'bios'
SER: directory /var/lock namestub LCK.. binary No
SER0: /dev/pts/1

but in a new log
Warning: SERIAL $Id$
registering /dev/pts/3 for serial
DPMI: mem init, mpool is 140468224 bytes at 0x7f3390800000
Warning: Using DPMI with CPU emulator

and later

This program is free software: you can redistribute it and/or modify^M
it under the terms of the GNU General Public License as published by^M
the Free Software Foundation, either version 3 of the License, or^M
(at your option) any later version.^M
^M
ERROR: SERIAL: Unable to open device /dev/pts/3: Permission denied
 - InitDisk: ^MC: HD1, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB^M

This unable to open device is something new...

**Reply by stsp** [reply]
Try `dosemu -no-priv-sep`.

**Comment by nelgin** [comment]
There you go, that fixed it. Thank you.

**Reply by stsp** [reply]
That fixed what?
Everything, or just the permission error?

**Reply by stsp** [reply]
And what is `/dev/pts/3`, is it
created by dosemu2, or by some
other prog?

**Comment by nelgin** [comment]
/dev/pts/3 is probably the tty device that spawns dosemu. In a shell if you sudo you get a new tty device.

CONF: time mode = 'bios'
SER: directory /var/lock namestub LCK.. binary No
SER0: /dev/pts/5

It looks like it's using it as the serial device?


bbs@bbs:/sbbs/exec$ tty
/dev/pts/1
bbs@bbs:/sbbs/exec$ sudo tty
/dev/pts/5

**Comment by nelgin** [comment]
Oh yes, that fixed everything. The games are now working again.

**Comment by stsp** [comment]
Moved to #2227
